controllers/operatorscaler_controller.go

```diff
@@ -0,0 +1,334 @@
+		return err
+	}
+
+	if odfDepsCsv.Status.Phase != oprv1a1.CSVPhaseSucceeded {
+		err = fmt.Errorf("csv %s is not in succeeded state", OdfDepsSubscriptionStartingCSV)
+		r.log.Error(err, "waiting for csv to be in succeeded state")
+		return err
+	}
+
+	return nil
+}
+
+func (r *OperatorScalerReconciler) reconcileOperators() error {
+
+	var returnErr error
+
+	for i := range ResourceMappingList {
+		resourceMapping := ResourceMappingList[i]
+
+		crList := &metav1.PartialObjectMetadataList{}
+		crList.TypeMeta.APIVersion = resourceMapping.ApiVersion
+		crList.TypeMeta.Kind = resourceMapping.Kind
+
+		err := r.Client.List(r.ctx, crList, client.Limit(1))
+		if err != nil {
+			if !meta.IsNoMatchError(err) {
+				msg := fmt.Sprintf("failed listing %s", resourceMapping.Kind)
+				r.log.Error(err, msg)
+				multierr.AppendInto(&returnErr, err)
+			}
+
+		} else if len(crList.Items) >= 0 {
+
+			for _, pkgName := range resourceMapping.PkgNames {
+				key := fmt.Sprintf(CsvLabelKey, pkgName, r.OperatorNamespace)
+
+				csvList := &oprv1a1.ClusterServiceVersionList{}
+				err = r.Client.List(
+					r.ctx, csvList,
+					client.InNamespace(r.OperatorNamespace),
+					client.MatchingLabels(map[string]string{key: ""}),
+				)
+				if err != nil {
+					r.log.Error(err, "failed listing csvs with label", "label", key)
+					multierr.AppendInto(&returnErr, err)
+				} else {
+					err = r.updateCsvDeplymentsReplicas(csvList.Items...)
+					if err != nil {
+						r.log.Error(err, "failed updating csvs replica")
+						multierr.AppendInto(&returnErr, err)
+					}
+				}
+			}
+		}
+	}
+
+	return returnErr
+}
+
+func (r *OperatorScalerReconciler) updateCsvDeplymentsReplicas(csvList ...oprv1a1.ClusterServiceVersion) error {
+
+	var returnErr error
+
+	for idx := range csvList {
+		csv := &csvList[idx]
+		var updateRequired bool
+		for i := range csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+			deploymentSpec := &csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs[i].Spec
+			if *deploymentSpec.Replicas < int32(1) {
+				deploymentSpec.Replicas = ptr.To(int32(1))
+				updateRequired = true
+			}
+		}
+
+		if updateRequired {
+			if err := r.Client.Update(r.ctx, csv); err != nil {
+				r.log.Error(err, "failed updating csv replica", "csvName", csv.Name)
+				multierr.AppendInto(&returnErr, err)
+			} else {
+				r.log.Info("csv updated successfully", "csvName", csv.Name)
+			}
+		}
+	}
+
+	return returnErr
+}
+
+func (r *OperatorScalerReconciler) reconcileDynamicWatchers() error {
+
+	for i := range ResourceMappingList {
+		resourceMapping := ResourceMappingList[i]
+
+		if !r.kindsBeingWatched[resourceMapping.Kind] {
+
+			crd := &extv1.CustomResourceDefinition{}
+			if err := r.Client.Get(r.ctx, types.NamespacedName{Name: resourceMapping.CrdName}, crd); err != nil {
+				if !errors.IsNotFound(err) {
+					r.log.Error(err, "failed getting crd", "crdName", resourceMapping.CrdName)
+					return err
+				}
+			} else {
```

Comment:
There is no need to for the added nesting. You can use this pattern instead
```suggestion
			crd := &extv1.CustomResourceDefinition{}
			crs.Name =  resourceMapping.CrdName
			if err := r.Client.Get(r.ctx, client.ObjectKeyFromObject(crs), crd); client.IgnoreNotFound(err) != nil {
				r.log.Error(err, "failed getting crd", "crdName", resourceMapping.CrdName)
				return err
			} else if err == nil {
```

Reply:
Done